packages/parchment/lib/src/document/embeds.dart

```diff
@@ -98,6 +98,7 @@ class BlockEmbed extends EmbeddableObject {
   }) : super(inline: false);
 
   static final BlockEmbed horizontalRule = BlockEmbed('hr');
-  static BlockEmbed image(String source) =>
-      BlockEmbed('image', data: {'source': source});
+
+  static BlockEmbed image(String source, {Map<String, dynamic>? data}) =>
+      BlockEmbed('image', data: (data ?? {})..['source'] = source);
```

Comment:
I just have a small comment: I prefer not to manipulate the data passed to a function, unless there's a valid reason for it and explicitly mentioned in the comments. So I prefer this version (https://github.com/fleather-editor/fleather/pull/497#discussion_r2187199558):
```dart
static BlockEmbed image(String source, Map<String, dynamic> data) =>
    BlockEmbed('image', data: {...data, 'source': source});
```